src/triage/predictlist/__init__.py

```diff
@@ -0,0 +1,593 @@
+        )
+        self.feature_generator.process_table_tasks(feature_aggregation_table_tasks)
+
+        # 4. Reconstruct feature disctionary from feature_names and generate imputation
+        feature_imputation_table_tasks = self.feature_generator.generate_all_table_tasks(
+            collate_aggregations,
+            task_type='imputation'
+        )
+        self.feature_generator.process_table_tasks(feature_imputation_table_tasks)
+        
+        feature_dict = self.feature_dictionary_creator.feature_dictionary(
+            feature_table_names=feature_imputation_table_tasks.keys(),
+            index_column_lookup=self.feature_generator.index_column_lookup(collate_aggregations),
+        )
+        feature_group_creator = FeatureGroupCreator({"all": [True]})
+        feature_group_mixer = FeatureGroupMixer(["all"])
+        feature_group_dict = feature_group_mixer.generate(
+            feature_group_creator.subsets(feature_dict) 
+        )[0]
+
+        # 5. Build new matrix
+        db_config = {
+            "features_schema_name": "triage_production",
+            "labels_schema_name": "public",
+            "cohort_table_name": cohort_table_name,
+            "labels_table_name": self.labels_table_name,
+        }
+
+        matrix_builder = MatrixBuilder(
+            db_config=db_config,
+            matrix_storage_engine=self.matrix_storage_engine,
+            engine=self.db_engine,
+            experiment_hash=None,
+            replace=True,
+        )
+        new_matrix_metadata = Planner.make_metadata(
+            matrix_definition=retrain_definition,
+            feature_dictionary=feature_group_dict,
+            label_name=self.label_name,
+            label_type='binary',
+            cohort_name=self.cohort_name,
+            matrix_type='train',
+            feature_start_time=self.feature_start_time,
+            user_metadata=self.user_metadata,
+        )
+        
+        new_matrix_metadata['matrix_id'] = "_".join(
+            [
+                self.label_name,
+                'binary',
+                str(as_of_date),
+                'retrain',
+                ]
+        )
+
+        matrix_uuid = filename_friendly_hash(new_matrix_metadata)
+        matrix_builder.build_matrix(
+            as_of_times=[as_of_date],
+            label_name=self.label_name,
+            label_type='binary',
+            feature_dictionary=feature_group_dict,
+            matrix_metadata=new_matrix_metadata,
+            matrix_uuid=matrix_uuid,
+            matrix_type="train",
+        )
+
+        misc_db_parameters = {
+            'train_end_time': dt_from_str(as_of_date),
+            'test': False,
+            'train_matrix_uuid': matrix_uuid, 
+            'training_label_timespan': self.training_label_timespan,
+            'model_comment': 'retrain_' + datetime.strftime(today, '%Y-%m-%d'),
+        }
+        retrained_model_id, retrained_model_hash = self.model_trainer._train_and_store_model(
```

Comment:
So here we're using an _-prefixed method, which as supposed to be quasi-private. It looks like the reason you're using this here (instead of the current public interface, `process_train_task` is to bypass the duplicate check. This by itself is fine (I can't think of why you would ever call the retrainer and not actually want to retrain the one model), but I think you could just do this by setting `replace=True` when instantiating the class and then call the existing public method `process_train_task`.

Reply:
yup you are right. made the change.